main.tf

```diff
@@ -81,7 +83,7 @@ data "aws_iam_policy_document" "origin_website" {
     condition {
       test     = "StringEquals"
       variable = "aws:referer"
-      values   = [random_password.referer.result]
+      values   = [random_password.referer[0].result]
```

Comment:
This will fail if `module.this.enabled = true` and `var.website_enabled = false` because `random_password.referer[0]` will not exist